azurerm/internal/services/recoveryservices/resource_arm_site_recovery_replicated_vm.go

```diff
@@ -161,14 +169,76 @@ func resourceArmSiteRecoveryReplicatedVM() *schema.Resource {
 					},
 				},
 			},
+			"network_interface": {
+				Type:       schema.TypeSet,
+				ConfigMode: schema.SchemaConfigModeAttr,
+				Computed:   true,
+				Optional:   true,
+				Elem:       netwrokInterfaceResource(),
+			},
+		},
+	}
+}
+
+func netwrokInterfaceResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"source_network_interface_id": {
+				Type:         schema.TypeString,
+				Computed:     true,
+				Optional:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+			"target_static_ip": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"target_subnet_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
 		},
 	}
 }
 
 func resourceArmSiteRecoveryReplicatedItemCreate(d *schema.ResourceData, meta interface{}) error {
-	name := d.Get("name").(string)
 	resGroup := d.Get("resource_group_name").(string)
 	vaultName := d.Get("recovery_vault_name").(string)
+
+	client := meta.(*clients.Client).RecoveryServices.ReplicationMigrationItemsClient(resGroup, vaultName)
+
+	err1 := initialCreation(d, client, meta)
+	if err1 != nil {
+		return err1
+	}
+
+	err2 := updateReplication(d, client, meta)
```

Comment:
This also means the resource will not return until replication is healthy where as before it would return before it does? what is the effect on the time to create the replication here? why do we need to update to finish the creation? could we document that in a comment

Reply:
To configure the NIC we need to wait for the replication. Replication usually take a long time, so we could save time and only wait if a NIC configuration is present. But then it would behave different if you choose to configure the NIC or not. For my usecase it is good that the VM is replicated once terraform returns but I understand that is not always the case. So I am open for suggestions here.